### PR TITLE
Fix for Python3 error in read.py (issue #162)

### PR DESCRIPTION
**Pull request by rupertford**

### CODE REVIEW

**Comment by rupertford** [comment]
I think this is ready for review by @arporter

File: src/fparser/scripts/read.py [hunk 1]
```diff
@@ -63,36 +64,56 @@
...
+
+def main():
+    '''Check input options then call the runner function.'''
     parser = OptionParser()
     set_read_options(parser)
-    if hasattr(parser, 'runner'):
-        parser.runner = runner
     options, args = parser.parse_args()
     runner(parser, options, args)
     return
```
Comment (arporter) [inline src/fparser/scripts/read.py]: pylint complains that this `return` is "useless" :-)

File: src/fparser/scripts/README [hunk 2]
```diff
@@ -0,0 +1,64 @@
+This directory contains scripts that are potentially useful for
+running and/or testing fparser. The scripts and their use is described
+below
+
+fparser2.py
+-----------
+
+Runs fparser2 on an input file and displays the de-parsed code. To see
+any command line options add -h. This script can be used to choose
+between parsing Fortran 2003 and Fortran 2008. An example of this
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/scripts/README]: OK, done. Also done for subsequent lines.

File: src/fparser/scripts/README [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+program test
+end program test
+> fparser2.py tmp.f90
+PROGRAM test
+END PROGRAM test
+
+parse.py
+--------
+
+Runs fparser1 on an input file and displays the parsers representation
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/scripts/README]: oop's ;-)

File: src/fparser/scripts/README [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    EndProgram
+      blocktype='program'
+      name='test'
+      item=Line('end program test',(2, 2),None,None,<reader>)
+
+read.py
+-------
+
+Runs the fparser file reader on an input file and displays the file
+readers representation of that code. This is useful for testing that
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/scripts/README]: whoops

File: src/fparser/scripts/read.py [hunk 5]
```diff
@@ -63,36 +64,56 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
 # DAMAGE.
 
-import os
+'''Python script with command line options which calls the Fortran
+File Reader with the supplied filename(s) and outputs the readers
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/scripts/read.py]: Well at least I'm consistent.

File: src/fparser/scripts/README [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+end program test
+> fparser2.py tmp.f90
+PROGRAM test
+END PROGRAM test
+
+parse.py
+--------
+
+Runs fparser1 on an input file and displays the parsers representation
+of that code. To see any command line options add -h. An example of
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/scripts/README]: Skipping this suggestion as agreed.

File: src/fparser/scripts/read.py [hunk 7]
```diff
@@ -63,36 +64,56 @@
...
+def runner(_, options, args):
+    '''Call the Fortran File reader for each filename in args and print
+    out its content.
 
-def runner (parser, options, args):
-    from fparser.readfortran import  FortranFileReader
+    :param options: command line argument information from the options \
+    parser
+    :type options: :py:class:`optparse.Values`
+    :param args: a list of fortran filepaths
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/fparser/scripts/read.py]: I don't mind picky if it is a simple fix :-)

File: src/fparser/scripts/read.py (hunk 1 above)
Comment (rupertford) [reply src/fparser/scripts/read.py]: My pylint (and an updated version) as well as pylint3 do not complain for some strange reason but your pylint is correct so I've removed it.

**Comment by rupertford** [comment]
I've hopefully addressed the reviewer's comments so ready for 2nd review from @arporter.

**Comment by arporter** [comment]
All comments have been addressed.

**Pull request merged**